chore(codemanagement) Lost and found PR's from develop

mariadb-LeonidFedorov opened the pull request:
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3335
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3025
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3160
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3168
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3174
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3228
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3234
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3244
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3249
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3252
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3256

mariadb-SergeyZefirov committed:
feat(primproc,aggregegation)!: Changes for ROLLUP with single-phase aggregation (#3025)

The fix is simple: enable subtotals in single-phase aggregation and
disable parallel processing when there are subtotals and aggregation is
single-phase.

mariadb-LeonidFedorov committed:
bug(priproc) make last_day type a bit more accurate

This fixes discrepance with the server, which assigns DATE type to
last_day()'s result.

Now we also assigns DATE result type and, also, use proper
dataconvert::Day data structure to return date.

Tests agree with InnoDB.

Also, this patch includes test for MCOL-5669, to show we fixed it.

denis0x0D committed:
fix(em): MCOL-5732 Add missing system OID to RebuildEM tool (#3168)

mariadb-SergeyZefirov committed:
feat(ddl): MCOL-5744: better handling of utf8 charset aliases (#3174)

Server expands ut8_XXX aliases to utf8mb3_XXX or utf8mb4_XXX depending
on the UTF8_IS_UTF8MB3 setting in the OLD_MODE environment variable.

Server already has the necessary code implemented in the get_utf8_flag()
method of class THD. There are several uses of this flag and all we have
to do to be in line with server is to use it.

This patch does that for DDL as work on MCOL-5705 uncovered some
problems in that area.

mariadb-SergeyZefirov committed:
fix(join): Fixes MCOL-5056, an error of joining TEXT column from InnoDB (#3160)

We incorrectly identified TEXT columns from external tables as BLOB.
Alexander Barkov suggested a way to discriminate them which I
implemented here.

mariadb-AlexanderPresniakov committed:
fix(funcexp): MCOL-4671 Fix behaviour of LEFT/RIGHT functions when negative trim length value is passedB

mariadb-SergeyZefirov committed:
fix: MCOL-5755: incorrect handling of BLOB (and TEXT) in GROUP BY

BLOB fields did not work as grouping keys at all, they were assigned
value NULL for any value, be it NULL or not. The fix is in the
rowaggregation.cpp in the initMapping(), a switch/case branch was added
to handle BLOB field copying there.

Also, TEXT columns did not distinguish between NULL and empty string in
the grouping algorithm, now they do. The fix is in the equals()
function, now we specifically check for isNull() equality between
values.

mariadb-AlexanderPresniakov committed:
fix(mcol-4499): Correct handling of LIKE/NOT LIKE NULL

mariadb-LeonidFedorov committed:
chore(codestyle): MCOL-5405: repace windows CRLF with virtious linux one

Commit:
fix(dbcon): MCOL-4756: having not() provokes an ERROR 2013

    The `NOT()` function in the HAVING clause was handled
    incorrectly, which caused the server to crash.

mariadb-AlexanderPresniakov committed:
Added more tests

mariadb-AlexeyAntipovsky committed:
fix(plugin): MCOL-5703 fix server crash on 'UNION ALL VALUES' queries

    Such queries are not supported by the ColumnStore anyway.

mariadb-AlexeyAntipovsky committed:
Add test case

mariadb-LeonidFedorov committed:
chore(build): fix duplicating hasRollup

mariadb-SergeyZefirov reviewed (approved):
I agree with code changes.

mariadb-LeonidFedorov committed:
MCOL-4671: MCOL-4622: fix the behavior of both PRs

first was playing different with RIGHT and LEFT functions(using the getUintVal and getIntVal accordingly)
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3234
second introduced round for ints from double, but added it to uint but not to int missing long doubles as well
https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3480

mariadb-LeonidFedorov committed:
Merge branch 'stable-23.10' into lost-and-found
